### PR TITLE
Refactor/delete wiremock

### PR DESCRIPTION
**Pull request by Eikix**
Time spent on this PR: 2d

Resolves: #305 

# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [x] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing

# What is the new behavior?

# Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: crates/core/src/client/errors.rs [hunk 1]
```diff
@@ -74,6 +74,12 @@ impl<T, E: std::error::Error> From<ConversionError<T>> for EthApiError<E> {
     }
 }
 
+impl<E: std::error::Error> From<FromByteSliceError> for EthApiError<E> {
```
Comment (Eikix) [inline crates/core/src/client/errors.rs:77]: boyscout commit to fix an issue discussed during a conversation about no more map_err

File: crates/core/src/client/mod.rs [hunk 2]
```diff
@@ -414,8 +414,7 @@ impl<P: Provider + Send + Sync> KakarotEthApi<P> for KakarotClient<P> {
         address: Address,
         contract_addresses: Vec<Address>,
     ) -> Result<TokenBalances, EthApiError<P::Error>> {
-        let entrypoint: Felt252Wrapper = keccak256("balanceOf(address)").try_into()?;
-        let entrypoint: FieldElement = entrypoint.into();
+        let entrypoint = FieldElement::from_byte_slice_be(&keccak256("balanceOf(address)").0[0..4])?;
```
Comment (Eikix) [inline crates/core/src/client/mod.rs:417]: boyscout commit to take only the 4 first bytes of a selector in EVM world

File: crates/core/src/client/tests/mod.rs [hunk 3]
```diff
@@ -1,44 +1,21 @@
...
     ABDEL_ETHEREUM_ADDRESS, ABDEL_STARKNET_ADDRESS, ABDEL_STARKNET_ADDRESS_HEX, ACCOUNT_ADDRESS, ACCOUNT_ADDRESS_EVM,
-    COUNTER_ADDRESS_EVM, INC_DATA, KAKAROT_ADDRESS, KAKAROT_TESTNET_ADDRESS, PROXY_ACCOUNT_CLASS_HASH,
-    PROXY_ACCOUNT_CLASS_HASH_HEX,
+    COUNTER_ADDRESS_EVM, INC_DATA, PROXY_ACCOUNT_CLASS_HASH_HEX,
 };
-use crate::mock::mock_starknet::{fixtures, mock_starknet_provider, AvailableFixtures, StarknetRpcFixture};
+use crate::mock::mock_starknet::{fixtures, init_mock_client, init_testnet_client, AvailableFixtures};
 use crate::wrap_kakarot;
 
-pub fn init_testnet_client() -> KakarotClient<SequencerGatewayProvider> {
```
Comment (Eikix) [inline crates/core/src/client/tests/mod.rs:25]: moving these two in mock folder, to be able to export them wherever we want

File: crates/core/src/mock/fixtures/responses/starknet_getTransactionByHash.json [hunk 4]
```diff
@@ -719,7 +719,7 @@
       "0x076e91a117d68549b7c7be395f1bd01596372f2ac631bd6ce6202430654434e",
       "0x04ef32bc4fd31910b365bff935637cc2b4a084c73a9bbd91e6f5e4fd6062deb0"
     ],
-    "transaction_hash": "0x03204b4c0e379c3a5ccb80d08661d5a538e95e2960581c9faf7ebcf8ff5a7d3c",
+    "transaction_hash": "0x3204b4c0e379c3a5ccb80d08661d5a538e95e2960581c9faf7ebcf8ff5a7d3c",
```
Comment (Eikix) [inline crates/core/src/mock/fixtures/responses/starknet_getTransactionByHash.json:722]: some issue here with leading zeroes, this isn't great, as I had to duplicate two routes solely because of leading zeroes, if you have a better way Id like to take it

File: crates/core/src/mock/mock_starknet.rs [hunk 5]
```diff
@@ -177,4 +235,10 @@ mod tests {
         assert_eq!(expected_params, fixture.params);
         assert_eq!(expected_response, fixture.response);
     }
+
+    #[test]
+    fn test_rpc_all_fixtures() {
+        let fixtures = all_fixtures();
+        assert_eq!(fixtures.len(), 22);
```
Comment (Eikix) [inline crates/core/src/mock/mock_starknet.rs:242]: not sure this is worth it -> if someone adds new fixtures, they'll have to update it

File: .vscode/settings.json [hunk 6]
```diff
@@ -3,5 +3,5 @@
   "files.exclude": {
     "**/.git": false
   },
-  "rust-analyzer.check.command": "clippy"
+  "rust-analyzer.check.command": "check"
```
Comment (ClementWalter) [inline .vscode/settings.json:6]: why this change?

File: crates/core/src/client/mod.rs (hunk 2 above)
Comment (ClementWalter) [reply crates/core/src/client/mod.rs:417]: this seems ok for now, but I think that we should use ethers-rs to do evm related manipulation. There is probably a `get_selector` or something similar

File: crates/core/src/mock/fixtures/requests/kakarot_getCode.json [hunk 7]
```diff
@@ -0,0 +1,13 @@
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "starknet_call",
+  "params": [
+    {
+      "contract_address": "0xd90fd6aa27edd344c5cbe1fe999611416b268658e866a54265aaf50d9cf28d",
```
Comment (ClementWalter) [inline crates/core/src/mock/fixtures/requests/kakarot_getCode.json:7]: I don't understand what is this request, why it's named `kakarot_getCode` while there is no calldata?

File: crates/core/src/mock/fixtures/requests/starknet_getEvents.json [hunk 8]
```diff
@@ -1,18 +0,0 @@
-{
-  "id": 1,
-  "jsonrpc": "2.0",
-  "method": "starknet_getEvents",
```
Comment (ClementWalter) [inline crates/core/src/mock/fixtures/requests/starknet_getEvents.json:4]: no more events testing?

File: crates/core/src/mock/fixtures/requests/starknet_getStorageAt.json [hunk 9]
```diff
@@ -1,12 +0,0 @@
-{
-  "id": 1,
-  "jsonrpc": "2.0",
-  "method": "starknet_getStorageAt",
```
Comment (ClementWalter) [inline crates/core/src/mock/fixtures/requests/starknet_getStorageAt.json:4]: no more storage testing?

File: crates/core/src/mock/mock_starknet.rs (hunk 5 above)
Comment (ClementWalter) [reply crates/core/src/mock/mock_starknet.rs:242]: not sure neither!

File: crates/core/Cargo.toml [hunk 10]
```diff
@@ -40,6 +40,7 @@ dotenv = { workspace = true }
 bytes = "1"  
 ethers = "2.0"
 foundry-config = { git = "https://github.com/foundry-rs/foundry", branch = "master" }  
+walkdir = "2.3.3"
```
Comment (ClementWalter) [inline crates/core/Cargo.toml:43]: need to remove wiremock as well

File: rpc-call-examples/getTransactionReceipt.hurl [hunk 11]
```diff
@@ -2,6 +2,6 @@ POST http://127.0.0.1:3030
 Content-Type: application/json
 {
     "jsonrpc":"2.0","method":"eth_getTransactionReceipt","params":[
-        "0x07d10a95dd633b4c84b3b9f04ebe380362898fc81779e7b016de6bd5b08deeae"
+        "0x06a78eeed9041c657b6f5c89dc6f5dbe17b463629d966ae807435278857e627b"
```
Comment (ClementWalter) [inline rpc-call-examples/getTransactionReceipt.hurl:5]: has this value a real meaning?

**Review by ClementWalter — COMMENTED** [review]
Looking at the deleted wiremock files, and considering that the wiremock looks like a standard accross diverse dev communities, I'm not sure to follow why the current setup with custom mocking is better

That being said, I've understood that we didn't use it anymore, so a cleaning is always welcome. I still would like to know what was wrong with it!

File: .vscode/settings.json (hunk 6 above)
Comment (Eikix) [reply .vscode/settings.json:6]: ahh just that my mac was suffering from clippy on save

File: crates/core/src/mock/fixtures/requests/starknet_getEvents.json (hunk 8 above)
Comment (Eikix) [reply crates/core/src/mock/fixtures/requests/starknet_getEvents.json:4]: they weren't being used yet

File: crates/core/src/mock/fixtures/requests/starknet_getStorageAt.json (hunk 9 above)
Comment (Eikix) [reply crates/core/src/mock/fixtures/requests/starknet_getStorageAt.json:4]: no used yet in testing suite

File: rpc-call-examples/getTransactionReceipt.hurl (hunk 11 above)
Comment (Eikix) [reply rpc-call-examples/getTransactionReceipt.hurl:5]: nope! will clean

File: .vscode/settings.json (hunk 6 above)
Comment (ClementWalter) [reply .vscode/settings.json:6]: wen new M2 max?

File: crates/core/src/mock/fixtures/requests/kakarot_getCode.json (hunk 7 above)
Comment (Eikix) [reply crates/core/src/mock/fixtures/requests/kakarot_getCode.json:7]: i think it is the evm_contract.bytecode() to fetch the bytecode of an evm contract on starknet, deployed thru kakarot

**Comment by Eikix** [comment]
> Looking at the deleted wiremock files, and considering that the wiremock looks like a standard accross diverse dev communities, I'm not sure to follow why the current setup with custom mocking is better
> 
> That being said, I've understood that we didn't use it anymore, so a cleaning is always welcome. I still would like to know what was wrong with it!

My feeling after having used wiremock on this project for the first months was that it was incredibly painful to set up routes, whereas the dojo mock client is easier to manipulate,
that being said, we should strive towards not using a mock client as much as possible but rather use the testsequencer for actual real interesting usage

File: crates/core/src/mock/mock_starknet.rs (hunk 5 above)
Comment (Eikix) [reply crates/core/src/mock/mock_starknet.rs]: deleted it

File: crates/core/src/client/mod.rs (hunk 2 above)
Comment (Eikix) [reply crates/core/src/client/mod.rs:417]: alright! will create an issue

File: crates/core/Cargo.toml (hunk 10 above)
Comment (Eikix) [reply crates/core/Cargo.toml:41]: done

File: crates/core/src/mock/fixtures/requests/kakarot_getCode.json (hunk 7 above)
Comment (greged93) [reply crates/core/src/mock/fixtures/requests/kakarot_getCode.json:7]: indeed the selector `0x2f22d9e1ae4a391b4a190b8225f2f6f772a083382b7ded3e8d85743a8fcfdcd` is the selector for bytecode

File: crates/core/src/mock/mock_starknet.rs [hunk 12]
```diff
@@ -141,6 +148,40 @@ pub fn fixtures(methods: Vec<AvailableFixtures>) -> Vec<StarknetRpcFixture> {
...
 
+/// Returns all the fixtures present in the fixtures directory.
+/// # Panics
+/// Returns an error if a fixture is not a valid json or if the fixture is not present in both
+/// requests and responses.
+///
+/// Panics if `all_fixtures` is called from a directory where `core` crate is not in scope.
+pub fn all_fixtures() -> Vec<StarknetRpcFixture> {
+    let cwd = &std::env::current_dir().unwrap();
+    let request_path = find_git_root_path(cwd).unwrap().join("crates/core/src/mock/fixtures/requests");
```
[the remark this answers is not in the document]
Comment (greged93) [reply crates/core/src/mock/mock_starknet.rs:159]: would it change anything to mark `mock` as `cfg(test)`?

[the remark this answers is not in the document]
Comment (Eikix) [reply crates/core/src/mock/mock_starknet.rs:159]: not sure! since we use it as a test-util, we export it to other crates

[the remark this answers is not in the document]
Comment (Eikix) [reply crates/core/src/mock/mock_starknet.rs:159]: i can try

[the remark this answers is not in the document]
Comment (Eikix) [reply crates/core/src/mock/mock_starknet.rs:159]: actually now I understand why dojo has a dedicated util crate, so that they can build it properly and then only import it as dev dependency to keep a lean binary size for their business logic

**Review by ClementWalter — APPROVED** [review]
